packages/odyssey-react-mui/src/labs/DataView/DataView.test.tsx

```diff
@@ -11,148 +11,993 @@
-        paginationType="loadMore"
-        resultsPerPage={20}
       />,
     );
 
-    const tableElement = await screen.findByRole("table", { name: "" });
-    const rowElements = within(tableElement).getAllByRole("row", {
-      hidden: false,
+    await waitFor(async () => {
+      const checkboxes = await screen.findAllByRole("checkbox");
+      expect(checkboxes.length).toBeGreaterThan(1);
+    });
+
+    const selectAllButton = screen.getByText("Select all", {
+      selector: "button",
+    });
+    fireEvent.click(selectAllButton);
+
+    expect(
+      await screen.findByText(`${simpleData.length} selected`),
+    ).toBeInTheDocument();
+
+    const selectNoneButton = screen.getByText("Select none", {
+      selector: "button",
     });
-    expect(rowElements.length).toBe(21);
+    fireEvent.click(selectNoneButton);
+
+    expect(await screen.queryByText("selected")).toBeNull();
+  });
+  it("can perform bulk actions on rows", async () => {
+    const bulkActionMenuItems = (selectedRows: MRT_RowSelectionState) => (
+      <MenuItem>Bulk action for {Object.keys(selectedRows).length}</MenuItem>
+    );
+
+    render(
+      <DataView
+        availableLayouts={["table"]}
+        getData={getSimpleData}
+        hasRowSelection
+        tableLayoutOptions={{
+          columns: columns,
+        }}
+        bulkActionMenuItems={bulkActionMenuItems}
+      />,
+    );
+
+    await waitFor(async () => {
+      const checkboxes = await screen.findAllByRole("checkbox");
+      expect(checkboxes.length).toBeGreaterThan(1);
+    });
+
+    const selectAllButton = screen.getByText("Select all", {
+      selector: "button",
+    });
+    fireEvent.click(selectAllButton);
+
+    const bulkActionsButton = await screen.findByText(
+      `${simpleData.length} selected`,
+    );
+    expect(bulkActionsButton).toBeInTheDocument();
+    fireEvent.click(bulkActionsButton);
+
+    const bulkActionMenu = await screen.findByRole("menuitem");
+    expect(bulkActionMenu.textContent).toBe(
+      `Bulk action for ${simpleData.length}`,
+    );
+  });
+
+  it("can reorder rows", async () => {
+    let data = [...simpleData];
+
+    const handleReorderRows = ({
+      rowId,
+      newRowIndex,
+    }: DataOnReorderRowsType) => {
+      data = reorderData({ data, rowId, newRowIndex });
+    };
+
+    render(
+      <DataView
+        availableLayouts={["table"]}
+        getData={() => data}
+        hasRowReordering
+        onReorderRows={handleReorderRows}
+        tableLayoutOptions={{
+          columns: columns,
+        }}
+      />,
+    );
+
+    await waitFor(async () => {
+      const rows = await screen.findAllByRole("row");
+      expect(rows.length).toBeGreaterThan(1);
+    });
+
+    const rows = await screen.findAllByRole("row");
+    expect(rows[1].textContent).toContain("Luke Skywalker");
+    expect(rows[2].textContent).toContain("Han Solo");
+
+    const reorderMenuButton = within(rows[2]).getAllByRole("button")[1];
```

Comment:
We can probably `getByLabel` here right? `aria-label` will be found with this method

Reply:
I see a similar selection method in some other places. If `getByLabel` works here as expected, we should update it elsewhere also